src/Storefront/Resources/app/storefront/src/plugin/captcha/google-re-captcha/google-re-captcha-base.plugin.js

```diff
@@ -2,6 +2,21 @@ import Plugin from 'src/plugin-system/plugin.class';
 
 export default class GoogleReCaptchaBasePlugin extends Plugin {
     init() {
+
+        // Ensure the script loading is initiated if data-src is present and src is not.
+        const recaptchaScript = document.getElementById('recaptcha-script');
+        if (recaptchaScript?.hasAttribute('data-src') && !recaptchaScript.getAttribute('src') && this._isValidUrl(recaptchaScript.getAttribute('data-src'))) {
+            recaptchaScript.setAttribute('src', encodeURI(recaptchaScript.getAttribute('data-src')));
+        }
```

Comment:
I always like to simplify these big conditions. Something like:

```suggestion
        const hasDataSrc = recaptchaScript?.hasAttribute('data-src');
        const srcAttr = recaptchaScript.getAttribute('src');
        const isValidUrl = this._isValidUrl(recaptchaScript.getAttribute('data-src'))

        if (hasDataSrc && !srcAttr && isValidUrl) {
            recaptchaScript.setAttribute('src', encodeURI(recaptchaScript.getAttribute('data-src')));
        }
```